include/clap/ext/draft/undo.h

```diff
@@ -0,0 +1,160 @@
+#pragma once
+
+#include "../../plugin.h"
+
+static CLAP_CONSTEXPR const char CLAP_EXT_UNDO[] = "clap.undo/2";
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/// @page Undo
+///
+/// This extension enables the plugin to merge its undo history with the host.
+/// This leads to a single undo history shared by the host and many plugins.
+///
+/// Calling host->undo() or host->redo() is equivalent to clicking undo/redo within the host's GUI.
+///
+/// If the plugin implements this interface then its undo and redo should be entirely delegated to
+/// the host; clicking in the plugin's UI undo or redo is equivalent to clicking undo or redo in the
+/// host's UI.
+///
+/// Some changes are long running changes, for example a mouse interaction will begin editing some
+/// complex data and it may take multiple events and a long duration to complete the change.
+/// In such case the plugin will call host->begin_change() to indicate the begining of a long
+/// running change and complete the change by calling host->change_made().
+///
+/// The host may group changes together:
+/// [---------------------------------]
+/// ^-T0      ^-T1    ^-T2            ^-T3
+/// Here a long running change C0 begin at T0.
+/// A instantaneous change C1 at T1, and another one C2 at T2.
+/// Then at T3 the long running change is completed.
+/// The host will then create a single undo step that will merge all the changes into C0.
+///
+/// This leads to another important consideration: starting a long running change without
+/// terminating is **VERY BAD**, because while a change is running it is impossible to call undo or
+/// redo.
+///
+/// Rationale: multiple designs were considered and this one has the benefit of having a single undo
+/// history. This simplifies the host implementation, leading to less bugs, a more robust design
+/// and maybe an easier experience for the user because there's a single undo context versus one
+/// for the host and one for each plugin instance.
+
+enum clap_undo_context_flags {
+   // While the host is within a change, it is impossible to perform undo or redo.
+   CLAP_UNDO_IS_WITHIN_CHANGE = 1 << 0,
+};
+
+enum clap_undo_delta_properties_flags {
+   // If not set, then all clap_undo_delta_properties's attributes becomes irrelevant.
+   // If set, then the plugin will provide deltas in host->change_made().
+   CLAP_UNDO_DELTA_PROPERTIES_HAS_DELTA = 1 << 0,
+
+   // If set, then the delta will be re-usable in the future as long as the plugin is
+   // compatible with the given format_version.
+   CLAP_UNDO_DELTA_PROPERTIES_IS_PERSISTANT = 1 << 0,
+};
```

Comment:
This constant has the same value as has delta 1<<0 I think it should be 1<<1

Reply:
Yes!!!

Reply:
Fixed